src/Microsoft.Azure.SignalR.Common/RestClients/SignalRServiceRestClient.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Http;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.SignalR
+{
+    public partial class SignalRServiceRestClient
+    {
+        private readonly string _userAgent;
+
+        public SignalRServiceRestClient(string userAgent, ServiceClientCredentials credentials, HttpClient httpClient, bool disposeHttpClient) : this(credentials, httpClient, disposeHttpClient)
```

Comment:
Is in our scenario userAgent always `ProductInfo.GetProductInfo(Assembly.GetCallingAssembly())`? If so, we don't need to expose this property

Reply:
We may want the assembly of function binding extension